flink-python/pyflink/datastream/state_backend.py

```diff
@@ -116,8 +119,280 @@ def __init__(self, j_state_backend):
+            for i in range(0, len(paths)):
+                j_path_array[i] = paths[i]
+            self._j_embedded_rocks_db_state_backend.setDbStoragePaths(j_path_array)
+
+    def get_db_storage_paths(self) -> List[str]:
+        """
+        Gets the configured local DB storage paths, or null, if none were configured.
+
+        Under these directories on the TaskManager, RocksDB stores its SST files and
+        metadata files. These directories do not need to be persistent, they can be ephermeral,
+        meaning that they are lost on a machine failure, because state in RocksDB is persisted
+        in checkpoints.
+
+        If nothing is configured, these directories default to the TaskManager's local
+        temporary file directories.
+
+        :return: The list of configured local DB storage paths.
+        """
+        return list(self._j_embedded_rocks_db_state_backend.getDbStoragePaths())
+
+    def is_incremental_checkpoints_enabled(self) -> bool:
+        """
+        Gets whether incremental checkpoints are enabled for this state backend.
+
+        :return: True if incremental checkpoints are enabled, false otherwise.
+        """
+        return self._j_embedded_rocks_db_state_backend.isIncrementalCheckpointsEnabled()
+
+    def set_predefined_options(self, options: 'PredefinedOptions'):
+        """
+        Sets the predefined options for RocksDB.
+
+        If user-configured options within ``RocksDBConfigurableOptions`` is set (through
+        flink-conf.yaml) or a user-defined options factory is set (via :func:`setOptions`),
+        then the options from the factory are applied on top of the here specified
+        predefined options and customized options.
+
+        Example:
+        ::
+
+            >>> state_backend.set_predefined_options(PredefinedOptions.SPINNING_DISK_OPTIMIZED)
+
+        :param options: The options to set (must not be null), see :class:`PredefinedOptions`.
+        """
+        self._j_embedded_rocks_db_state_backend\
+            .setPredefinedOptions(options._to_j_predefined_options())
+
+    def get_predefined_options(self) -> 'PredefinedOptions':
+        """
+        Gets the current predefined options for RocksDB.
+        The default options (if nothing was set via :func:`setPredefinedOptions`)
+        are :data:`PredefinedOptions.DEFAULT`.
+
+        If user-configured options within ``RocksDBConfigurableOptions`` is set (through
+        flink-conf.yaml) or a user-defined options factory is set (via :func:`setOptions`),
+        then the options from the factory are applied on top of the predefined and customized
+        options.
+
+        .. seealso:: :func:`set_predefined_options`
+
+        :return: Current predefined options.
+        """
+        j_predefined_options = self._j_embedded_rocks_db_state_backend.getPredefinedOptions()
+        return PredefinedOptions._from_j_predefined_options(j_predefined_options)
+
+    def set_options(self, options_factory_class_name: str):
```

Comment:
I guess it's not enough to only pass the class name of the options_factory. For example, for DefaultConfigurableOptionsFactory, there are many methods in it to configure it.

Reply:
No, this is fine (it's actually copied directly from RocksDBStateBackend). This is just loading a factory class that is defined in Java. We could think about making the factory definable in Python but that is a separate issue.